turbopack/crates/turbo-tasks-backend/src/backend/operation/mod.rs

```diff
@@ -165,10 +165,20 @@ where
         category: TaskDataCategory,
     ) -> Vec<CachedDataItem> {
         // Safety: `transaction` is a valid transaction from `self.backend.backing_storage`.
-        unsafe {
+        let result = unsafe {
             self.backend
                 .backing_storage
                 .lookup_data(self.transaction(), task_id, category)
+        };
+        match result {
+            Ok(data) => data,
+            Err(e) => {
+                let task_name = self.backend.get_task_description(task_id);
+                panic!(
+                    "Failed to restore task data (corrupted database or bug): {:?}",
```

Comment:
should we instruct the user to delete the cache and report an issue?

bgw referenced a plan to automatically invalidate the db at this point... add a reference?

Reply:
He is probably doing that on any panic...